### PR TITLE
updated probe handler types descriptions

### PR DESCRIPTION
**Pull request by SergeyKanzhelev**
Follow up of https://github.com/kubernetes/kubernetes/pull/105979/

#### What type of PR is this?

Comments added may not be needed as the consistency long term will be guaranteed by back compatibility requirement.

Also removed todo and false statements.

/kind cleanup
/sig node
/priority backlog
/label api-review

```release-note
NONE
```

### CODE REVIEW

**Comment by SergeyKanzhelev** [comment]
cc: @thockin

File: pkg/apis/core/types.go [hunk 1]
```diff
@@ -2189,37 +2189,31 @@ type Container struct {
 }
 
 // ProbeHandler defines a specific action that should be taken in a probe.
-// This type has a strong relationship to LifecycleHandler - overlapping fields
-// should be identical.
-// TODO: pass structured data to these actions, and document that data here.
```
Comment (SergeyKanzhelev) [inline pkg/apis/core/types.go:2194]: I don't think we will do it any time soon, removing TODO for clarity

File: pkg/apis/core/types.go [hunk 2]
```diff
@@ -2189,37 +2189,31 @@ type Container struct {
...
+// One and only one of the following should be specified.
```
Comment (SergeyKanzhelev) [inline pkg/apis/core/types.go:2192]: moved this text here as otherwise it become a comment of an Exec probe only.

File: pkg/apis/core/types.go [hunk 3]
```diff
@@ -2189,37 +2189,31 @@ type Container struct {
 }
 
 // ProbeHandler defines a specific action that should be taken in a probe.
-// This type has a strong relationship to LifecycleHandler - overlapping fields
```
Comment (SergeyKanzhelev) [inline pkg/apis/core/types.go:2192]: there is no need to keep this information. Backward compatibility requirements will take care of this

File: pkg/apis/core/types.go [hunk 4]
```diff
@@ -2189,37 +2189,31 @@ type Container struct {
...
-	// One and only one of the following should be specified.
 	// Exec specifies the action to take.
 	// +optional
 	Exec *ExecAction
 	// HTTPGet specifies the http request to perform.
 	// +optional
 	HTTPGet *HTTPGetAction
-	// TCPSocket specifies an action involving a TCP port.
-	// TODO: implement a realistic TCP lifecycle hook
+	// Deprecated. TCPSocket is NOT supported as a LifecycleHandler and kept
```
Comment (SergeyKanzhelev) [inline pkg/apis/core/types.go:2214]: followed example of some other fields. Interestingly, Deprecated with dot does not actually mark the field as deprecated. Deprecated with semicolon does, but makes generated code fail the static check.

File: pkg/apis/core/types.go [hunk 5]
```diff
@@ -2235,9 +2229,8 @@ type Lifecycle struct {
 	// PreStop is called immediately before a container is terminated due to an
 	// API request or management event such as liveness/startup probe failure,
 	// preemption, resource contention, etc. The handler is not called if the
-	// container crashes or exits. The reason for termination is passed to the
```
Comment (SergeyKanzhelev) [inline pkg/apis/core/types.go:2238]: > The reason for termination is passed to the handler.

I don't know what this is referring to. I didn't find evidence that this is true or was true at some point

**Review by SergeyKanzhelev — COMMENTED** [review]
added comments for reviewers

File: pkg/apis/core/types.go (hunk 3 above)
Comment (thockin) [reply pkg/apis/core/types.go:2192]: This was forward-looking - if we use the same name it should probably be the same schema?  Maybe not,  but I assumed so...

File: pkg/apis/core/types.go (hunk 2 above)
Comment (thockin) [reply pkg/apis/core/types.go:2192]: Maybe "One and only one field should be specified." ?

File: pkg/apis/core/types.go [hunk 6]
```diff
@@ -2189,37 +2189,31 @@ type Container struct {
...
-	// TODO: implement a realistic TCP lifecycle hook
 	// +optional
 	TCPSocket *TCPSocketAction
 }
 
 // LifecycleHandler defines a specific action that should be taken in a lifecycle
-// hook.  This type has a strong relationship to ProbeHandler - overlapping fields
-// should be identical.
-// TODO: pass structured data to these actions, and document that data here.
+// hook. One and only one of the fields, except TCPSocket should be specified.
```
Comment (thockin) [inline pkg/apis/core/types.go:2206]: Should have the same "one and only one" comment?

File: pkg/apis/core/types.go (hunk 4 above)
Comment (thockin) [reply pkg/apis/core/types.go:2214]: What error?  We should fix that.

File: pkg/apis/core/types.go [hunk 7]
```diff
@@ -2235,9 +2229,8 @@ type Lifecycle struct {
 	// PreStop is called immediately before a container is terminated due to an
 	// API request or management event such as liveness/startup probe failure,
 	// preemption, resource contention, etc. The handler is not called if the
-	// container crashes or exits. The reason for termination is passed to the
-	// handler. The Pod's termination grace period countdown begins before the
-	// PreStop hooked is executed. Regardless of the outcome of the handler, the
+	// container crashes or exits. The Pod's termination grace period countdown begins before the
+	// PreStop hook is executed. Regardless of the outcome of the handler, the
 	// container will eventually terminate within the Pod's termination grace
 	// period. Other management of the container blocks until the hook completes
```
Comment (thockin) [inline pkg/apis/core/types.go:2235]: Add a couple words on finalizers here?

**Review by thockin — COMMENTED** [review]
LGTM overall

File: pkg/apis/core/types.go (hunk 3 above)
Comment (SergeyKanzhelev) [reply pkg/apis/core/types.go:2192]: hm, this is guaranteed by the matching types like `ExecAction`.  The only thing this comment help with is the presence of fields, which is also guaranteed by back-compat requirement.

File: pkg/apis/core/types.go (hunk 6 above)
Comment (SergeyKanzhelev) [reply pkg/apis/core/types.go:2206]: hm, I don't understand where it's missing.

File: pkg/apis/core/types.go (hunk 4 above)
Comment (SergeyKanzhelev) [reply pkg/apis/core/types.go:2214]: `./hack/verify-staticcheck.sh`:

```
Errors from staticcheck:
pkg/apis/core/v1/zz_generated.conversion.go:4127:2: out.TCPSocket is deprecated: TCPSocket is NOT supported as a LifecycleHandler and kept for the backward compatibility. There are no validation of this field and lifecycle hooks will fail in runtime when tcp handler is specified. +optional  (SA1019)
pkg/apis/core/v1/zz_generated.conversion.go:4139:55: in.TCPSocket is deprecated: TCPSocket is NOT supported as a LifecycleHandler and kept for the backward compatibility. There are no validation of this field and lifecycle hooks will fail in runtime when tcp handler is specified. +optional  (SA1019)
pkg/apis/core/validation/validation.go:2697:14: lh.TCPSocket is deprecated: TCPSocket is NOT supported as a LifecycleHandler and kept for the backward compatibility. There are no validation of this field and lifecycle hooks will fail in runtime when tcp handler is specified. +optional  (SA1019)
```

Comment (SergeyKanzhelev) [reply pkg/apis/core/types.go:2214]: maybe we can exclude generated files from the static check?

File: pkg/apis/core/types.go (hunk 7 above)
Comment (SergeyKanzhelev) [reply pkg/apis/core/types.go:2235]: what do you have in mind? Just reference them?

File: pkg/apis/core/types.go (hunk 3 above)
Comment (thockin) [reply pkg/apis/core/types.go:2192]: I'm going to resolve it.  I think I am wrong

File: pkg/apis/core/types.go (hunk 2 above)
Comment (thockin) [reply pkg/apis/core/types.go:2192]: Specifically "of the following" is weird outside the struct

File: pkg/apis/core/types.go (hunk 6 above)
Comment (thockin) [reply pkg/apis/core/types.go:2206]: I don't either.  Sorry

File: pkg/apis/core/types.go (hunk 4 above)
Comment (thockin) [reply pkg/apis/core/types.go:2214]: yes, I think that's in the works.  Can you file a cleanup bug to come fix this after staticcheck changes get merged?

File: pkg/apis/core/types.go (hunk 7 above)
Comment (thockin) [reply pkg/apis/core/types.go:2235]: Something like "will eventually terminate within the Pod's termination grace period (unless delayed by finalizers). " ?

Comment (SergeyKanzhelev) [reply pkg/apis/core/types.go]: thank you!

File: pkg/apis/core/types.go (hunk 4 above)
Comment (SergeyKanzhelev) [reply pkg/apis/core/types.go]: https://github.com/kubernetes/kubernetes/issues/106405

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Comment by thockin** [comment]
Removed LGTM

Is the goal to make these changes user-visible?  The equivalent changes need to be made in staging/src/k8s.io/api/... and then a million hack/update-* need to run.

**Comment by thockin** [comment]
Thanks!

/lgtm
/approve

File: pkg/apis/core/types.go (hunk 5 above)
Comment (cbandy) [reply pkg/apis/core/types.go:2238]: Fixes #99250.

**Pull request merged**